lmdeploy/serve/openai/api_server.py

```diff
@@ -334,6 +335,13 @@ async def chat_completions_v1(request: ChatCompletionRequest,
     error_check_ret = await check_request(request)
     if error_check_ret is not None:
         return error_check_ret
+    try:
+        _ = ChatCompletionRequestMessage(messages=request.messages)
```

Comment:
we can just replace the type hint https://github.com/irexyc/lmdeploy/blob/cc142150780351aebd9dac00539cd74e76d7a369/lmdeploy/serve/openai/protocol.py#L113 . It will check if it is valid itself.

Reply:
The messages will converted to list of class if replace the type hint, however the messages2prompt function needs the messages to be list of dict.